src/workerd/jsg/filesystem.c++

```diff
@@ -0,0 +1,537 @@
+#include "filesystem.h"
+
+#include "buffersource.h"
+
+namespace workerd::jsg::fs {
+
+namespace {
+constexpr size_t kMaxReadLength = 32 * 1024;
+
+void validatePathWithNoSeparators(kj::StringPtr path) {
+  try {
+    auto parsed = kj::Path::parse(path);
+    JSG_REQUIRE(parsed.size() == 1, Error, "Invalid path: \"", path, "\"");
```

Comment:
I'm not sure I'm following, what does this validate?

Reply:
Validates that the path has only a single component, e.g. `foo` and not `foo/bar`, etc

Reply:
Ah, I see. Maybe renaming the function to `validatePathHasSingleComponent` is better?

Reply:
I'm not sure that renaming it to that would really be much of an improvement. Added comments